packages/app/contexts/FathomAnalyticsContext.tsx

```diff
@@ -0,0 +1,51 @@
+"use client";
+
+import {
+  createContext,
+  PropsWithChildren,
+  useContext,
+  useEffect,
+  useMemo,
+} from "react";
+
+import { EventName } from "@/constants";
+import { EventOptions } from "@/analytics";
+
+const siteId = process.env.NEXT_PUBLIC_FATHOM_SITE_ID;
+
+interface AnalyticsContextProps {
+  trackClick: (eventName: EventName, opts?: EventOptions) => void;
+}
+
+const AnalyticsContext = createContext<AnalyticsContextProps>({
+  trackClick: () => {
+    throw new Error("No AnalyticsContext available");
+  },
+});
+
+export const AnalyticsProvider = ({ children }: PropsWithChildren) => {
+  useEffect(() => {
+    if (!siteId) throw new Error("Site ID not set, skipping Fathom analytics");
+  }, []);
+
+  const analyticsContextValues = useMemo(
+    () => ({
+      trackClick: (eventName: EventName, opts?: EventOptions) => {
+        window.fathom.trackEvent(eventName, opts);
+        console.log(
+          `Click event emitted: ${eventName}.${opts ? " Options: " : ""}`,
```

Comment:
I don't think we should log here.

Reply:
It's just a util function to know which event are we triggering but I can remove it, honestly, I don't have strong feelings about it.
Removing it now :D